src/pages/fi-opening-balances/forms/FiOpeningBalancesForm.js

```diff
@@ -0,0 +1,222 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import { useForm } from 'src/hooks/form'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+
+export default function FiOpeningBalancesForms({ labels, maxAccess, recordId }) {
+  const [isLoading, setIsLoading] = useState(false)
+  const [editMode, setEditMode] = useState(!!recordId)
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.FiOpeningBalance.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      accountId: null,
+      fiscalYear: null,
+      amount: null,
+      baseAmount: null,
+      currencyId: null,
+      plantId: null,
+      plantName: '',
+      plantRef:''
+    },
+    maxAccess: maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      fiscalYear: yup.string().required(' '),
+      accountId: yup.string().required(' '),
+      currencyId: yup.string().required(' '),
+      amount: yup.string().required(' '),
+      baseAmount: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+
+      const response = await postRequest({
+        extension: FinancialRepository.FiOpeningBalance.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!recordId) {
+        toast.success('Record Added Successfully')
+        formik.setValues({
+          ...obj,
+          recordId: response.recordId 
+        })
+      } else toast.success('Record Edited Successfully')
+      setEditMode(true)
+
+      invalidate()
+    }
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+          setIsLoading(true)
+
+          const res = await getRequest({
+            extension: FinancialRepository.FiOpeningBalance.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          formik.setValues(res.record)
+        }
+      } catch (exception) {
+        setErrorMessage(error)
```

Comment:
remove this also. any error in the API requests are handled in the requests context